### DISCUSSION TITLE
Allow to disable joypad support

### DISCUSSION THREAD

**Discussion by timothyqiu** [godotengine/godot-proposals#8390] [Engine Core]
It's convenient to use a joypad as a shortcut keyboard via tools like xpadder and JoyToKey. It's popular for things like drawing and audio/video editing.

The problem is that switching to Godot always captures the joypad. To setup the joypad as a keyboard again, one has to disconnect the joypad, close Godot, then reconnect the joypad.

I suggest allowing to disable joypad support at runtime. Or at least adding a `--disable-joypad` command-line argument to disable it when the engine starts.

**Comment by zhyohaGit** [comment]
This is a very good suggestion, and I really hope that it can be supported in future versions of Godot.

**Comment by Calinou** [comment]
This could also be a workaround for https://github.com/godotengine/godot/issues/20566, so it makes sense to provide a CLI argument for this. Many other engines like Source provide a similar CLI argument.

**Comment by jsjtxietian** [comment]
Should we add a project setting for this? Maybe users may have trouble providing a custon CLI argument for godot, but they can open the project, do the setting and then restart the editor.

**Reply by timothyqiu** [reply]
Both a project setting (to always disable for the exported project) and an editor setting (to always disable for the editor) I think.

**Comment by timothyqiu** [comment]
Created a dedicated proposal for this: #8391
